lua/entities/gmod_wire_expression2/core/string.lua

```diff
@@ -395,7 +395,9 @@ end
 --- Finds and replaces every occurrence of <needle> with <new> without regular expressions
 e2function string string:replace(string needle, string new)
 	if needle == "" then return this end
-	return this:Replace( needle, new)
+	local ret = this:Replace(needle, new)
+	self.prf = self.prf + #ret * 0.3 + #self * 0.1 + #needle * 0.1 + #string * 0.1
```

Comment:
You should do this before the operation, check if prf exceeds the max and error if it does.

Reply:
i did it this way using explode() and explodeRE() as a reference, those ones check after the operation too

Reply:
Could you make it like this? https://github.com/wiremod/wire/blob/master/lua/entities/gmod_wire_expression2/core/string.lua#L315-L317

Reply:
ah, okay, i see now
fixed in latest commit